transforms/universal/ededup/python/src/ededup_transform_python.py

```diff
@@ -50,25 +58,89 @@ def _process_cached_hashes(self, hd: dict[str, str]) -> list[str]:
             unique.append(hd[uh])
         return unique
 
-    def flush(self) -> tuple[list[pa.Table], dict[str, Any]]:
+
+class EdedupPythonRuntime(DefaultPythonTransformRuntime):
+    """
+    Exact dedup runtime support
+    """
+
+    def __init__(self, params: dict[str, Any]):
+        super().__init__(params=params)
+        self.filter = None
+
+    def get_transform_config(
+        self, data_access_factory: DataAccessFactoryBase, statistics: TransformStatistics, files: list[str]
+    ) -> dict[str, Any]:
         """
-        We are implementing flush here to compute additional statistics
-        :return: additional metadata
+        Get the dictionary of configuration that will be provided to the transform's initializer.
+        This is the opportunity for this runtime to create a new set of configuration based on the
+        config/params provided to this instance's initializer.  This may include the addition
+        of new configuration data such as ray shared memory, new actors, etc., that might be needed and
+        expected by the transform in its initializer and/or transform() methods.
+        :param data_access_factory - data access factory class being used by the RayOrchestrator.
+        :param statistics - reference to statistics actor
+        :param files - list of files to process
+        :return: dictionary of transform init params
+        """
+        if self.params.get("use_snapshot", False):
+            snapshot_path = self.params.get("snapshot_directory", None)
+            if snapshot_path is None or len(snapshot_path) == 0:
+                snapshot_path = (
+                    f"{SnapshotUtils.get_snapshot_folder(data_access_factory.create_data_access())}"
+                    f"hash_collector_1"
```

Comment:
Is it ok to hardcode "hash_collector_1" and why the "_1"?  Perhaps some documentation on the structure of the snapshot directory would be useful - reading code to figure out is not nice.

Reply:
Because in the case of Python, there is only 1